experimental/cloudstrapper/playbooks/roles/agw-config-update/tasks/main.yaml

```diff
@@ -0,0 +1,87 @@
+---
+
+- name: Check if rootCA exists
+  stat:
+    path: "{{ agwCertsPath }}/rootCA.pem"
+  become: true
+  register: has_rootca
+  failed_when:
+    - not has_rootca.stat.exists
+
+- name: Check if control_proxy.yml exists
+  stat:
+    path: "{{ agwConfigPath }}/control_proxy.yml"
+  become: true
+  register: has_control_proxy
+  failed_when:
+    - not has_control_proxy.stat.exists
+
+# Reference: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/lineinfile_module.html
+- name: Check if cloud_address is correctly configured in control_proxy
+  lineinfile:
+    state: absent
+    path: "{{ agwConfigPath }}/control_proxy.yml"
+    regex: '^cloud_address:[ ]{1,}controller.{{ orc8rDomainName }}'
+  check_mode: true
+  become: true
+  register: check_cloud_address
+  changed_when: false # Do not count this as a change, it is just a check
+
+- name: Configure cloud_address in control_proxy
+  lineinfile:
+    path: "{{ agwConfigPath }}/control_proxy.yml"
+    regex: '^cloud_address:.*$'
```

Comment:
Really like how you look for a pattern and change it only if it's not the case. However, if the errant pattern does not match this regex above, we would not be adding the right string here at all. Would it help if we add an error string here that says "no pattern matching ^cloud_address, incorrect control_proxy.yml"? Same applies for the others as well.